src/Drivers/Gd/Modifiers/TrimModifier.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Intervention\Image\Drivers\Gd\Modifiers;
+
+use Intervention\Image\Drivers\DriverSpecialized;
+use Intervention\Image\Exceptions\NotSupportedException;
+use Intervention\Image\Interfaces\ImageInterface;
+use Intervention\Image\Interfaces\ModifierInterface;
+
+/**
+ * @property int $tolerance
+ */
+class TrimModifier extends DriverSpecialized implements ModifierInterface
+{
+    public function apply(ImageInterface $image): ImageInterface
+    {
+        if ($image->isAnimated()) {
+            throw new NotSupportedException('Animated Images do not support trim operation');
+        }
+
+        $gd      = $image->core()->native();
+        $width   = imagesx($image->core()->native());
+        $height  = imagesy($image->core()->native());
+        $red     = 0;
+        $green   = 0;
+        $blue    = 0;
+        $corners = [
+            [0,0],
+            [$width - 1,0],
+            [0,$height - 1],
+            [$width - 1,$height - 1],
+        ];
+
+        foreach ($corners as $corner) {
+            $thisColor = imagecolorat($gd, $corner[0], $corner[1]);
+
+            $rgb    = imagecolorsforindex($gd, $thisColor);
+            $red   += round(round(($rgb['red'] / 0x33)) * 0x33);
+            $green += round(round(($rgb['green'] / 0x33)) * 0x33);
+            $blue  += round(round(($rgb['blue'] / 0x33)) * 0x33);
```

Comment:
Just curious. Why do you use `0x33` instead of dividing directly by 51?

Reply:
Because RGB values are in hexadecimal notation. I thought it made sense 😅.